app/search/facetSearch.js

```diff
@@ -22,96 +54,53 @@ app.controller('FacetSearchCtrl', function($scope, $location, searchService, ont
     current: 1
   };
 
-  function getResultsPage() {
-    if(isTermSearch) {
-      $scope.queryPromise = searchService.findTerms($scope.searchTerm, $scope.maxSize, $scope.currentPage, facets,
+  function sortAndTrimFacets(fields) {
+    _.each(fields, function(facet) {
+      // _.each($scope.results.facet.facetFields, function(facet) {
+      facet.categories = _.sortBy(facet.categories, function(category) {
+        return category.count;
+      });
+      facet.categories = _.last(facet.categories, 10).reverse();
+    });
+  }
+
+  function getResultsPage(type) {
+    if(type === 'term') {
+      $scope.queryPromise = searchService.findTerms($scope.searchTerm, $scope.maxSize, $scope.currentPage, _.pluck(facets,'id'),
           $scope.filters);
     } else {
       $scope.queryPromise = searchService.findGeneProducts($scope.searchTerm, $scope.maxSize, $scope.currentPage,
-          facets, $scope.filters);
+          _.pluck(facets,'id'), $scope.filters);
     }
     $scope.queryPromise.then(
       function(result) {
         $scope.results = result.data;
-        postProcess();
+        sortAndTrimFacets($scope.results.facet.facetFields);
+
+        var taxaIds = _.uniq(_.pluck(result.data.results, 'taxonId'));
+        if(type === 'gp') {
+          taxaIds = taxaIds.concat(_.pluck(_.find($scope.results.facet.facetFields, function(d){
+              return d.field === 'taxonId';
+          }).categories,'name'));
+          taxonomyService.getTaxa(taxaIds).then(function(response){
+            angular.forEach(response.data.taxonomies, function(taxon) {
+              $scope.taxaMapping[taxon.taxonomyId] = taxon.scientificName;
+            });
+          });
+        }
       });
   }
 
   $scope.pageChanged = function() {
-    getResultsPage();
-  };
-
-  $scope.isGoTerm = function(termId) {
-    return ontoTypeService.isGoTerm(termId);
+    getResultsPage(type);
   };
 
-  $scope.highlight = function(text) {
+  $scope.highlight = function(text) {//TODO check this is still working
```

Comment:
The styling was missing. To avoid this kind of issue in the future, I've just added italics to highlight the word.